### PR TITLE
Clone dataset schema

### PR DESCRIPTION
**Pull request by hinthornw**

### CODE REVIEW

File: python/langsmith/client.py [hunk 1]
```diff
@@ -3283,6 +3284,8 @@ def clone_public_dataset(
                 dataset_name=dataset_name,
                 description=ds.description,
                 data_type=ds.data_type or ls_schemas.DataType.kv,
+                inputs_schema=ds.inputs_schema,
```
Comment (jakerachleff) [inline python/langsmith/client.py:3287]: can we set transformations as well here?

**Review by jakerachleff — CHANGES_REQUESTED** [review]

File: python/langsmith/client.py [hunk 2]
```diff
@@ -2806,22 +2807,34 @@ def create_dataset(
         ----------
         dataset_name : str
             The name of the dataset.
-        description : str or None, default=None
+        description : Optional[str], default=None
             The description of the dataset.
-        data_type : DataType or None, default=DataType.kv
+        data_type : ls_schemas.DataType, default=ls_schemas.DataType.kv
```
Comment (jakerachleff) [inline python/langsmith/client.py:2812]: note to self to mark this as deprecated

File: python/langsmith/schemas.py [hunk 3]
```diff
@@ -171,6 +171,22 @@ class Config:
         frozen = True
 
 
+DatasetTransformationType = Literal[
```
Comment (jakerachleff) [inline python/langsmith/schemas.py:174]: Any thoughts on not making this a literal so it's easier to add new ones server side?

**Review by jakerachleff — APPROVED** [review]

File: python/langsmith/schemas.py (hunk 3 above)
Comment (hinthornw) [reply python/langsmith/schemas.py:174]: I think since I union with str it at least won't be blocked if you try to say a new one

**Pull request merged**